watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/dye_desalination/dye_desalination.py

```diff
@@ -148,71 +215,216 @@ def eq_flow_mass_comp(blk, j):
+        prtrt.wwtp = SecondaryTreatmentWWTPZO(
+            property_package=m.fs.prop_nf, database=m.db, process_subtype="default"
+        )
+    else:
+        pass
+
+    # nanofiltration components
+    dye_sep.P1 = PumpElectricityZO(
+        property_package=m.fs.prop_nf, database=m.db, process_subtype="default"
+    )
+
+    dye_sep.nanofiltration = NanofiltrationZO(
+        property_package=m.fs.prop_nf,
+        database=m.db,
+        process_subtype="rHGO_dye_rejection",
+    )
+
+    # reverse osmosis components
+    if include_RO:
+        desal = m.fs.desalination = Block()
+        desal.P2 = Pump(property_package=m.fs.prop_ro)
+        if RO_1D:
+            desal.RO = ReverseOsmosis1D(
+                property_package=m.fs.prop_ro,
+                has_pressure_change=True,
+                pressure_change_type=PressureChangeType.calculated,
+                mass_transfer_coefficient=MassTransferCoefficient.calculated,
+                concentration_polarization_type=ConcentrationPolarizationType.calculated,
+            )
+        else:
+            desal.RO = ReverseOsmosis0D(
+                property_package=m.fs.prop_ro,
+                has_pressure_change=True,
+                pressure_change_type=PressureChangeType.calculated,
+                mass_transfer_coefficient=MassTransferCoefficient.calculated,
+                concentration_polarization_type=ConcentrationPolarizationType.calculated,
+            )
+
+        desal.RO.width.setub(2000)
+        desal.RO.area.setub(20000)
+
+        desal.S1 = Separator(property_package=m.fs.prop_ro, outlet_list=["P2", "PXR"])
+        desal.M1 = Mixer(
+            property_package=m.fs.prop_ro,
+            momentum_mixing_type=MomentumMixingType.equality,
+            inlet_list=["P2", "P3"],
+        )
+        desal.PXR = PressureExchanger(property_package=m.fs.prop_ro)
+        desal.P3 = Pump(property_package=m.fs.prop_ro)
+
+        m.fs.permeate = Product(property_package=m.fs.prop_ro)
+        m.fs.brine = Product(property_package=m.fs.prop_ro)
+
+        # translator blocks
+        m.fs.tb_nf_ro = Translator(
+            inlet_property_package=m.fs.prop_nf, outlet_property_package=m.fs.prop_ro
+        )
+
+        # since the dye << tds: Assume RO_TDS = NF_tds + NF_dye
+        @m.fs.tb_nf_ro.Constraint(["H2O", "dye"])
+        def eq_flow_mass_comp(blk, j):
+            if j == "dye":
+                return (
+                    blk.properties_in[0].flow_mass_comp["dye"]
+                    + blk.properties_in[0].flow_mass_comp["tds"]
+                    == blk.properties_out[0].flow_mass_phase_comp["Liq", "TDS"]
+                )
+            else:
+                return (
+                    blk.properties_in[0].flow_mass_comp["H2O"]
+                    == blk.properties_out[0].flow_mass_phase_comp["Liq", "H2O"]
+                )
+
     else:
-        m.fs.dye_retentate = Product(property_package=m.fs.prop)
+        pass
```

Comment:
Is this unnecessary?

Reply:
The else statement isn't necessary, but I just figured it was good practice. I can get rid of this and similar "pass" else statements if we'd like.

Reply:
I think this can stay in. The reason being that it helps with readability.